docs/learn-more/unreachable-commits.md

```diff
@@ -0,0 +1,70 @@
+# Reachable and Unreachable Commits
+
+In Git, every commit will have at least one parent commit except the very first. Additionally, a repository may have any number of branches that begin at any particular commit. Because of this we can create a graph of the history of a commit by following the path from one commit's parent to the another. Given a branch `development`, whose initial commit is `A`, and we add commit `B` and `C`. A resulting graph would be as follows:
+
+```
+development
+C
+|
+B
+|
+A
+```
+
+Since we can follow the graph from `C` to `A`, that means that `A` is **reachable** by or from `C`. This as known as following the ancestral path of `C`.
+
+## Unreachable Commits
+Now, if we create a new branch called `feature-branch` from `C` and commit `D` and `E` and then return to `development` and commit `F`. We would have the resulting graph:
+```
+development
+F
+|  E
+|  |  <-- feature-branch
+|  D
+| /
+C
+|
+B
+|
+A
+```
+
+In the above example, `B` is reachable by `F` through the ancestral path of `F`-> `C` -> `B`  and `B` is reachable by `E` through `E` -> `D` -> `C` -> `B`. However, there is no such path to get to `E` or `D` from `F`. Thus, `E` and `D` are **unreachable** from `F`.
+
+## Git Commands Use the Ancestral Path
+Some git commands use the ancestral path to determine what to show. One of those is `git diff`, which is used to see the changes between two commits non inclusive of the first commit. If we execute `git diff A..C`, we will receive the set of changes from the commits along the ancestral path from `C` to `A` or `C` -> `B` -> `A`. Thus, we would see changes from `B` and `C`. Likewise, if we executed `git diff B..F`, you will get changes from reachable from `F`; thus, `F` -> `C` -> `B`. But, not changes from `E` and `D` as they are unreachable.
+
+## Merge Commits
+Now, lets say we merge the `feature-branch` into our `development` branch. Our graph becomes:
+```
+development
+G
+| \
+F  |
+|  E
+|  |  <-- feature-branch
+|  D
+| /
+C
+|
+B
+|
+A
+```
```

Comment:
```mermaid
    gitGraph
       commit id: "A"
       commit id: "B"
       commit id: "C"
       branch feature-branch
       checkout feature-branch
       commit id: "D"
       commit id: "E"
       checkout main
       commit id: "F"
       merge feature-branch tag: "G"
```

(there is no `id:` for the `merge` command, so I used a tag instead 😢 the `id` support for merge commits should be WIP: https://github.com/mermaid-js/mermaid/issues/3238#event-7215873738)